### ISSUE TITLE
Default psychopy prefs are overriding expected audioLib selection

### ISSUE THREAD

**Issue by mwaskom** [psychopy/psychopy#1529]
The sound module [docs](https://github.com/psychopy/psychopy/blob/master/psychopy/sound/__init__.py#L3) suggest that `sounddevice` is a supported and prioritized audiolibrary. However, when I have only `sounddevice` installed, I cannot import `psychopy.sound`:

```
import psychopy.sound
```
```
DependencyError: No sound libs could be loaded. Tried: [u'pyo', u'pygame']
```

This list is coming from `psychopy.prefs.general["audioLib"]`. I have no custom user preferences file (as reported by `psychopy.prefs.resetPrefs()`), and cannot figure out where this list of libraries is being populated and why it's missing `sounddevice`.

I understand that I can override the list in my script (or I guess make a config file) but my question is whether the default behavior is broken.

**Comment by CRiddler** [comment]
It does certainly appear that the default behavior isn't functioning properly. I'll take a more in-depth look at this tomorrow when I have time during the day. Can you comment the version of psychopy you're running?

**Comment by mwaskom** [comment]
This is on the latest PyPI release.

**Comment by CRiddler** [comment]
Thanks!

That was quite the chase to track down, but it seems that our `preferences/baseNoArch.spec` files (essentially where psychopy gets the preferences from if there is no "userPrefs.cgf" file). I was able to re-create this issue and resolve it as follows:

	>>> import psychopy
	>>> psychopy.prefs.general['audioLib']
	[u'sounddevice', u'pyo', u'pygame']

	#Use resetPrefs() deletes my userPrefs.cfg file and loads the prefs from the 
	  #appropriate (OS specific) `.spec` file
	>>> psychopy.prefs.resetPrefs()

	#My audioLibs are now short 1 lib!
	>>> psychopy.prefs.general['audioLib']
	[u'pyo', u'pygame']

	#reset again to show that I actually have NO userPrefs.cfg file any more
	>>> psychopy.prefs.resetPrefs()
	Could not remove prefs file '/Users/riddellcd/.psychopy2/userPrefs.cfg'; (try doing it manually?)

	#Still short 1 lib
	psychopy.prefs.general['audioLib']
	[u'pyo', u'pygame']

    ##########
    The Patch:
	#First go inside of `psychopy/preferences and edit `baseNoArch.spec` (L38 on my file)
	  #The line should currently read:
	  audioLib = list(default=list('pyo', 'pygame'))
	  #Change it to
	  audioLib = list(default=list('sounddevice', 'pyo', 'pygame'))
	#Now we need to run the generateSpec code- since its not wrapped in `if __name__ == '__main__',
	#importing it will run it for us- this script reads in `baseNoArch.spec` and recreates all of the 
	#OS specific `.spec` files
	from psychopy.preferences import generateSpec

	#Reload my prefs from the `.spec` file via `resetPrefs()`
	psychopy.prefs.resetPrefs()
	Could not remove prefs file '/Users/riddellcd/.psychopy2/userPrefs.cfg'; (try doing it manually?)

	#My prefs now have sounddevice in my 'audioLib'
	psychopy.prefs.general['audioLib']
	[u'sounddevice', u'pyo', u'pygame']

@peircej just checking in to make sure that editing the `.spec` file in this fashion is acceptable to do as I'm not terribly familiar with this end of psychopy. If it is, we should update `baseNoArch.spec` to include sounddevice.

@mwaskom let me know if you have any questions or if this doesn't work for you! This was a tricky one to track down, so I'm hoping its the root of your issue :)

**Comment by hoechenberger** [comment]
Hi @CRiddler, I'm not Jon ;) But yes, `baseNorArch.spec` seems to be the right place to edit. Be sure to run `generateSpec.py` to generate the derived OS-specific files after that, and include the them in the PR as well. Good work!

**Comment by CRiddler** [comment]
I saw you had already added some tags so I figured you were watching this issue ;) thanks for the response! I ran `generateSpec.py` in my "fix" by importing it since it had no `if __name__ == '__main__' I'll have a quick PR submitted by this afternoon! Just have some things at work to do first :(

**Comment by hoechenberger** [comment]
> I ran generateSpec.py in my "fix" by importing it since it had no `if __name__ == __main__` 

I don't quite understand... the easiest way is to run this script from the command line, i.e., 
`python generateSpec.py`

> I'll have a quick PR submitted by this afternoon! Just have some things at work to do first :(

Don't worry, no-one's in a rush here. And no-one's paying you to hurry (or even to to _anything_, at all), anyway ;)

**Comment by CRiddler** [comment]
> I don't quite understand... the easiest way is to run this script from the command line, i.e.,

Ah yes, I meant easiest as in easiest to show since I was doing this inside of the python interpreter the whole time. However, it probably would've been better to actually call the script from the command line since its more explicit as to what I'm specifically doing. :)

**Comment by peircej** [comment]
I think that's the right thing to do; Jeremy came up with the system of using `base` and then `generateSpec.py` as a way to make sure the different architectures have the same.

With regards to the use of sounddevice lib, it had been something that we had left for people to turn on manually (i.e. they should go to their prefs and choose it rather than us choosing it for them, because it was new (and therefore not for newbies). But I think it's fair to add in to the defaults now; it's been a while.

Thanks @CRiddler (and @hoechenberger is spot on with his advice on all points above)

**Comment by mwaskom** [comment]
As it wasn't obvious to me at first, this is closed with #1531
